monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/core/data/manager/InMemoryDataManager.java

```diff
@@ -0,0 +1,238 @@
+
+    public InMemoryDataManager() {
+        heartBeatStore = new HashMap<>();
+        serviceInfoStore = new HashMap<>();
+        logger.debug("heartBeatStore and serviceInfoStore created ");
+    }
+
+    @Override
+    public boolean insertHeartbeat(HeartbeatObject heartbeat, String accessToken) {
+        try {
+            HashMap map = new HashMap<>();
+            map.put("groupId", heartbeat.getGroupId());
+            map.put("nodeId", heartbeat.getNodeId());
+            map.put("interval", heartbeat.getInterval());
+            map.put("mgtUrl", heartbeat.getMgtApiUrl());
+            map.put("timeStamp", heartbeat.getTimestamp());
+            map.put("accessToken", accessToken);
+            String keyString = heartbeat.getGroupId() + heartbeat.getNodeId();
+            heartBeatStore.put(keyString, map);
+            logger.info("Inserted heartbeat to node " + heartbeat.getNodeId() +  " of group " + heartbeat.getGroupId());
+            return heartBeatStore.size() > 0;
+        } catch (DashboardServerException e) {
+            throw new DashboardServerException("Error occurred while inserting heartbeat information.", e);
+        }
+    }
+
+    @Override
+    public boolean insertServerInformation(HeartbeatObject heartbeat, String serverInfo) {
+        try {
+            HashMap map = new HashMap();
+            map.put("groupId", heartbeat.getGroupId());
+            map.put("nodeId", heartbeat.getNodeId());
+            map.put("serviceInfo", serverInfo);
+            String keyString = heartbeat.getGroupId() + heartbeat.getNodeId();
+            serviceInfoStore.put(keyString, map);
+            logger.info("Added serverInfo to node " + heartbeat.getNodeId() +  " of group " + heartbeat.getGroupId());
+            return serviceInfoStore.size() > 0;
+        } catch (DashboardServerException e) {
+            throw new DashboardServerException("Error occurred while inserting server information of node : "
+                                               + heartbeat.getNodeId() + " in group: " + heartbeat.getGroupId(), e);
+        }
+    }
+
+    @Override
+    public GroupList fetchGroups() {
+        String groupId;
+        try {
+            GroupList groupList = new GroupList();
+            for (HashMap entry: heartBeatStore.values()) {
+                groupId = entry.get("groupId").toString();
+                if (!groupList.contains(groupId)) {
+                    groupList.add(groupId);
+                }
+            }
+            return groupList;
+        } catch (DashboardServerException e) {
+            throw new DashboardServerException("Error occurred fetching groups.", e);
+        }
+    }
+
+    @Override
+    public NodeList fetchNodes(String groupId) {
+        try {
+            NodeList nodeList = new NodeList();
+            for (HashMap entry: serviceInfoStore.values()) {
+                if (entry.get("groupId").toString().equals(groupId)) {
```

Comment:
Can the entry become null here. If so we need to check that before evaluate.

Reply:
entry may not be null here.